storage/storage.go

```diff
@@ -175,3 +175,44 @@ type notImplemented struct {
 func (e notImplemented) Error() string {
 	return fmt.Sprintf("%q is not supported on %q storage", e.method, e.apiType)
 }
+
+type Metadata map[string]string
+
+func (m Metadata) SetACL(acl string) Metadata {
+	m["ACL"] = acl
+	return m
+}
+
+func (m Metadata) ACL() string {
+	acl, ok := m["ACL"]
+	if !ok {
+		return ""
+	}
+	return acl
+}
+
+func (m Metadata) SetStorageClass(class string) Metadata {
+	m["StorageClass"] = class
+	return m
+}
+
+func (m Metadata) StorageClass() string {
+	stclass, ok := m["StorageClass"]
+	if !ok {
+		return ""
+	}
+	return stclass
+}
+
+func (m Metadata) SetContentType(contentType string) Metadata {
```

Comment:
what do you think of removing these set methods since it is not currently used? once we need them in the future, we may add them.

Reply:
sounds logical, thanks.